File: app/assets/javascripts/models/family.js
App.Family = DS.Model.extend({
  // attributes
  name: DS.attr('string'),
  color: DS.attr('string'),
  kind: DS.attr('string'),

  // relations
  social_network: DS.belongsTo('App.SocialNetwork'),
  nodes: DS.hasMany('App.Node'),

  // computed properties
  isActorKind: function() {
    return this.get('kind') === "Actor";
  }.property('kind'),
});

File: app/models/family.rb
class Family < ActiveRecord::Base
  belongs_to :social_network
  has_and_belongs_to_many :nodes
  attr_accessible :color, :kind, :name, :social_network_id, :node_ids

  def uri(base = social_network.prefix)
    @uri ||= social_network.uri(base) + "/families/#{id}"
  end
end
